### ISSUE TITLE
Doesn't account for redirects.

### ISSUE THREAD

**Issue by andrewfluck** [hinok/react-router-last-location#17]
Using the library, I noticed this doesn't account for any redirects.

For example I'm on `/` and goto `/profile` but I get redirected to `/profile/hosting`.

I have a back button that does `goBack()` which goes back to `/` but I'd like the link to reflect that which only reflects `/profile`.

If I wasn't clear the flow should be:
`/` -> `/profile` redirect to `/profile/hosting` -> `/`
**not**
`/` -> `/profile` redirect to `/profile/hosting` -> `/profile`
for the `to` prop.

**Comment by hinok** [comment]
Hey @andrewfluck ,
Good point, many thanks for reporting the issue! I'll look into it.

**Comment by hinok** [comment]
@andrewfluck 

Finally I've had some free time to think about this problem and how it could be solved.
In my opinion `<Redirect />` and `this.props.history.replace()` can be `rendered/called` some time after fully loading a page so using just one approach for all redirects in an application doesn't seem good for me.

**Example**
User visits a success page like `/success` that contains such code:

```js
setTimeout(() => {
  this.props.history.replace('/');
}, 5000);
```

Some people may want to store `/success` as last location, others may not. Not every page or component will render `<Redirect />` immediately (server-side redirects !== client-side redirects).

My proposal to fix redirects that shouldn't notify `react-router-last-location` is by using `location.state`, like below:

```jsx
import React from 'react';
import { Redirect } from 'react-router-dom';

const FixedRedirect = () => (
  <Redirect to={{ pathname: '/', state: { preventLastLocation: true } }} />
);

export default FixedRedirect;
```

I'm keen to hear your opinion.

**Comment by andrewfluck** [comment]
So when you pass in `preventLastLocation` to your redirect state, the library will act as it did before? Or is it the other way around?

**Comment by hinok** [comment]
When you pass in `preventLastLocation` to your redirect state, the library **will not store** that "redirected" route as `lastLocation`.

**Example**

```jsx
<Route path="/" component={Home} />
<Route path="/about" component={About} />
<Route path="/redirect" component={RedirectUser} />

const RedirectUser = () => <Redirect path="/" />
```

If user navigates:

- Visit homepage `/`, `lastLocation = null`
- Visit `/about`, `lastLocation = { pathname: "/", ... }`
- Visit `/redirect`, `lastLocation = { pathname: "about", ... }`
- Redirected to `/`, `lastLocation = { pathname: "about", ... }`

**Comment by andrewfluck** [comment]
What if you were to make it a global thing in the context provider as a prop as well? That way, it wouldn't need to be passed around everywhere I redirect?

**Comment by hinok** [comment]
You meant something like

**A)**

```jsx
import { RedirectWithoutLastLocation } from 'react-router-last-location';

// code

return <RedirectWithoutLastLocation to="/" />
```

or something like

**B)**

```jsx
// code
const { lastLocation, Redirect } = useLastLocation();

return <Redirect to="/" />
```

At the moment context/provider is only used for `useLastLocation` hook. I like more **A)** but both of proposals don't cover a case when someone would like to call directly `props.history.push / replace`.

If you could prepare a small snippet of code how API should look in your opinion, that would be helpful.

**Comment by andrewfluck** [comment]
I would include both tbh.

**Comment by rafecolton** [comment]
@hinok this feature would be great! Is it slated for release any time soon?

**Comment by hinok** [comment]
@rafecolton I'll try to push this today or if I find bugs, tomorrow or end of this week.

**Comment by rafecolton** [comment]
Awesome, thank you!

**Comment by hinok** [comment]
@rafecolton Published `2.0.0` that contains `RedirectWithoutLastLocation`, example + usage is in README.md

https://github.com/hinok/react-router-last-location#use-redirectwithoutlastlocation-to-dont-store-redirects-as-last-location

**Comment by rafecolton** [comment]
@hinok thank you, thank you!
